genai-perf/genai_perf/inputs/retrievers/synthetic_prompt_generator.py

```diff
@@ -174,6 +174,12 @@ def _generate_prompt_with_token_reuse(
         """
         final_prompt: List[int] = []
         size_to_use = block_size
+        last_hash_length = num_tokens - ((len(prompt_hash_list) - 1) * block_size)
+        assert 0 < last_hash_length <= block_size, (
+            f"Input_length: {num_tokens}, Hash_ids: {prompt_hash_list}, Block_size: {block_size} "
```

Comment:
Could you make this error message tell the user what the accept input range is for the number of hash blocks they're requesting? And how that is calculated? We want this to be intuitive so that users know what the error is when they get it.

Also, we don't really use asserts in our code. We raise errors that gets caught by GenAI-Perf in main. Can you do it that way or is there a specific reason to diverge and use an assert here?

Reply:
Updated to make it an exception.